projects/packages/jetpack-mu-wpcom/src/features/wpcom-block-editor/class-jetpack-wpcom-block-editor.php

```diff
@@ -0,0 +1,634 @@
+					'wp-blocks',
+					'wp-data',
+					'wp-dom-ready',
+					'wp-plugins',
+				),
+				$version,
+				true
+			);
+			wp_enqueue_style(
+				'wpcom-block-editor-wpcom-editor-styles',
+				$debug
+					? '//widgets.wp.com/wpcom-block-editor/wpcom.editor.css?minify=false'
+					: '//widgets.wp.com/wpcom-block-editor/wpcom.editor.min.css',
+				array(),
+				$version
+			);
+		}
+
+		if ( $this->is_iframed_block_editor() ) {
+			wp_enqueue_script(
+				'wpcom-block-editor-calypso-editor-script',
+				$debug
+					? '//widgets.wp.com/wpcom-block-editor/calypso.editor.js?minify=false'
+					: '//widgets.wp.com/wpcom-block-editor/calypso.editor.min.js',
+				array(
+					'calypsoify_wpadminmods_js',
+					'jquery',
+					'lodash',
+					'react',
+					'wp-blocks',
+					'wp-data',
+					'wp-hooks',
+					'wp-tinymce',
+					'wp-url',
+				),
+				$version,
+				true
+			);
+
+			wp_enqueue_style(
+				'wpcom-block-editor-calypso-editor-styles',
+				$debug
+					? '//widgets.wp.com/wpcom-block-editor/calypso.editor.css?minify=false'
+					: '//widgets.wp.com/wpcom-block-editor/calypso.editor.min.css',
+				array(),
+				$version
+			);
+		}
+	}
+
+	/**
+	 * Enqueues the WordPress.com block editor integration assets for both editor and front-end.
+	 */
+	public function enqueue_block_assets() {
+		// These styles are manually copied from //widgets.wp.com/wpcom-block-editor/default.view.css in order to
+		// improve the performance by avoiding an extra network request to download the CSS file on every page.
+		wp_add_inline_style( 'wp-block-library', '.has-text-align-justify{text-align:justify;}' );
+	}
+
+	/**
+	 * Determines if the current $post contains a justified paragraph block.
+	 *
+	 * @return boolean true if justified paragraph is found, false otherwise.
+	 */
+	public function has_justified_block() {
+		global $post;
+		if ( ! $post instanceof WP_Post ) {
+			return false;
+		}
+
+		if ( ! has_blocks( $post ) ) {
+			return false;
+		}
+
+		return str_contains( $post->post_content, '<!-- wp:paragraph {"align":"justify"' );
+	}
+
+	/**
+	 * Register the Tiny MCE plugins for the WordPress.com block editor integration.
+	 *
+	 * @param array $plugin_array An array of external Tiny MCE plugins.
+	 * @return array External TinyMCE plugins.
+	 */
+	public function add_tinymce_plugins( $plugin_array ) {
+		if ( $this->is_iframed_block_editor() ) {
+			$debug = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG;
+
+			$plugin_array['gutenberg-wpcom-iframe-media-modal'] = add_query_arg(
+				'v',
+				gmdate( 'YW' ),
+				$debug
+					? '//widgets.wp.com/wpcom-block-editor/calypso.tinymce.js?minify=false'
+					: '//widgets.wp.com/wpcom-block-editor/calypso.tinymce.min.js'
+			);
+		}
+
+		return $plugin_array;
+	}
+
+	/**
+	 * Ensures the authentication cookies are designated for cross-site access.
+	 */
+	private function enable_cross_site_auth_cookies() {
+		/**
+		 * Allow plugins to disable the cross-site auth cookies.
+		 *
+		 * @since 8.1.1
```

Comment:
I think we'll want to update all `@since` references  here and below to `jetpack-`, to clearly indicate that this was originally from the Jetpack plugin.

```suggestion
		 * @since jetpack-8.1.1
```

Reply:
Done in [97f89df](https://github.com/Automattic/jetpack/pull/37324/commits/97f89dfa1fa2996054f85e6117047f5a686aff33)